src/libs/actions/IOU.ts

```diff
@@ -3442,40 +3572,38 @@ function cancelPayment(expenseReport, chatReport) {
                 statusNum: CONST.REPORT.STATUS_NUM.REIMBURSED,
             },
         },
-        ...(chatReport.reportID
-            ? [
-                  {
-                      onyxMethod: Onyx.METHOD.MERGE,
-                      key: `${ONYXKEYS.COLLECTION.REPORT}${chatReport.reportID}`,
-                      value: {
-                          hasOutstandingIOU: false,
-                          hasOutstandingChildRequest: false,
-                          iouReportID: 0,
-                      },
-                  },
-              ]
-            : []),
     ];
 
-    API.write(
-        'CancelPayment',
-        {
-            iouReportID: expenseReport.reportID,
-            chatReportID: chatReport.reportID,
-            managerAccountID: expenseReport.managerID,
-            reportActionID: optimisticReportAction.reportActionID,
-        },
-        {optimisticData, successData, failureData},
-    );
+    if (chatReport?.reportID) {
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${chatReport.reportID}`,
+            value: {
+                hasOutstandingIOU: false,
```

Comment:
```suggestion
```

Reply:
Same, it exists in the main, so it's not something added by me and I'm not sure if I should remove it during migration
https://github.com/Expensify/App/blob/5b03db96e5a2816ba18cd5757f076037807baa90/src/libs/actions/IOU.js#L3450-L3454